src/platforms/ios/iosiaphandler.h

```diff
@@ -28,6 +31,12 @@ class IOSIAPHandler final : public PurchaseIAPHandler {
  private:
   void* m_delegate = nullptr;
   int discountToDays(void* discount);
+  // This is a void (and cast to InAppPurchaseHandler as needed) as the
+  // InAppPurchaseHandler needs to store Product, which is only available in iOS
+  // 15+. So the entire class must be marked as only available in iOS 15. But we
+  // can't mark this variable as only available in iOS 15+ here. Hence, we have
+  // this variable as a `void`, and cast it to InAppPurchaseHandler as needed.
+  void* swiftIAPHandler;
```

Comment:
Can we simply use a forward class declaration of in this case? IIUC, it's fine if the type doesn't get defined so long as it's used as a pointer. eg: 

```
class InAppPurchaseHandler;

class IOSIAPHandler {
    // ... some stuff ...
    InAppPurchaseHandler* swiftIAPHandler = nullptr;
};
```

Reply:
Oh, and we should also provide an initial value here of `nullptr`

Reply:
Good call on nullptr, added.

We can't use `InAppPurchaseHandler` here because the `InAppPurchaseHandler` class is marked as `@available(iOS 15, *)` at the top. That means in iOS 14, the app doesn't know about the class - even the name. (It seemed to make sense to mark the entire class as needing 15+ rather than marking nearly every function in the class, but I'm open to other thoughts. It felt most in line with the intent of availability marking.) However, we can't mark things as available in C++, so we're in a bit of a weird spot. (I tried to explain this in the note above this line as it's nonstandard, though if it's not clear please let me know.)